File: src/db-backend/test-programs/javascript/javascript_hcr_flow_test/mymodule.js
// mymodule.js -- version 1 (pre-reload)

function compute(n) {
    // v1: doubles the input
    return n * 2;
}

function transform(value, n) {
    // v1: adds n to value
    return value + n;
}

function aggregate(history) {
    // v1: sums the history array
    return history.reduce(function(a, b) { return a + b; }, 0);
}

module.exports = { compute: compute, transform: transform, aggregate: aggregate };

File: src/db-backend/test-programs/ruby/ruby_hcr_flow_test/mymodule.rb
# mymodule.rb -- version 1 (pre-reload)

def compute(n)
  # v1: doubles the input
  n * 2
end

def transform(value, n)
  # v1: adds n to value
  value + n
end

def aggregate(history)
  # v1: sums the history array
  history.sum
end
